plugins/uefi/fu-plugin-uefi.c

```diff
@@ -135,6 +137,202 @@ fu_plugin_get_results (FuPlugin *plugin, FuDevice *device, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_plugin_uefi_update_resource (fwup_resource *re,
+				guint64 hardware_instance,
+				GBytes *blob,
+				GError **error)
+{
+	int rc;
+	rc = fwup_set_up_update_with_buf (re, hardware_instance,
+					  g_bytes_get_data (blob, NULL),
+					  g_bytes_get_size (blob));
+	if (rc < 0) {
+		g_autoptr(GString) str = g_string_new (NULL);
+		rc = 1;
+		for (int i = 0; rc > 0; i++) {
+			char *filename = NULL;
+			char *function = NULL;
+			char *message = NULL;
+			int line = 0;
+			int err = 0;
+
+			rc = efi_error_get (i, &filename, &function, &line,
+					    &message, &err);
+			if (rc <= 0)
+				break;
+			g_string_append_printf (str, "{error #%d} %s:%d %s(): %s: %s, ",
+						i, filename, line, function,
+						message, strerror (err));
+		}
+		if (str->len > 2)
+			g_string_truncate (str, str->len - 2);
```

Comment:
What's this truncating?  Is strerror(err) adding in two junk characters every time?

Reply:
I changed the error message to add `, ` at the end of the string; we don't have any other multi-line error messages in fwupd but I can change that back if you think it's best.

Reply:
Almost always there will be a few error messages that come back at once and it's really difficult to visually parse if the only thing separating them is commas.  So I think this should either be a multi-line or a more obvious delimiter.  Maybe a pipe?

Reply:
A tab? Double tab?

Reply:
Tab would work well enough I think.